dpd/src/api_server.rs

```diff
@@ -3061,6 +3070,200 @@ async fn counter_get(
         .map_err(HttpError::from)
 }
 
+#[derive(Deserialize, Serialize, JsonSchema)]
+pub struct MulticastGroupIdParam {
+    pub group_id: String,
+}
+
+// Implement conversion to mcast::Identifier
+impl MulticastGroupIdParam {
+    pub fn to_identifier(&self) -> Result<mcast::Identifier, HttpError> {
+        // First try to parse as IP address
```

Comment:
Why not just model `MulticastGroupIdParam` as an enum in the first place?

Reply:
yeah. the api_server has some stuff defined only in API vs jsonschema/serd/de on the structures themselves, but I starting going in that direction, so I agree here.

Reply:
I'm re-arranging this, but now I remember why: dropshot doesn't allow it on the param side.